src/openms_gui/source/VISUAL/SpectrumCanvas.cpp

```diff
@@ -391,60 +391,120 @@ namespace OpenMS
 
   bool SpectrumCanvas::addLayer(ExperimentSharedPtrType map, ODExperimentSharedPtrType od_map, const String & filename)
   {
-    layers_.resize(layers_.size() + 1);
-    layers_.back().param = param_;
-    layers_.back().filename = filename;
-    layers_.back().setPeakData(map);
-    layers_.back().setOnDiscPeakData(od_map);
-
-    if (layers_.back().getPeakData()->getChromatograms().size() != 0 
-        && layers_.back().getPeakData()->size() != 0)
+    LayerData new_layer;
+    new_layer.param = param_;
+    new_layer.filename = filename;
+    new_layer.name = QFileInfo(filename.toQString()).baseName();
+    new_layer.setPeakData(map);
+    new_layer.setOnDiscPeakData(od_map);
+
+    if (new_layer.getPeakData()->getChromatograms().size() != 0 
```

Comment:
use `if (!new_layer.getPeakData()->getChromatograms().empty() && !new_layer.getPeakData().getSpectra()->empty()` for clarity?